compiler/semmagic.nim

```diff
@@ -575,7 +575,27 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
     result = n
   of mPrivateAccess:
     var t = n[1].typ[0]
-    if t.kind in {tyRef, tyPtr}: t = t[0]
+    #[
+    find the underlying `object`, even in cases like these:
+    type
+      B[T] = object f0: int
+      A1[T] = ref B[T]
+      A2[T] = ref object f1: int
+      A3 = ref object f2: int
+      A4 = object f3: int
+    ]#
+    var foundPtrLike = false
+    while true:
+      case t.kind
+      of tyRef, tyPtr:
+        if foundPtrLike: break
+        foundPtrLike = true
```

Comment:
Btw I think it should follow all pointers much like we deref them all for `objRefRefRef.field`

Reply:
done; 
I didn't know derefs are more than 1 level, is that documented? (=> just filed https://github.com/nim-lang/Nim/issues/18298)